stake-pool/program/src/instruction.rs

```diff
@@ -418,24 +421,27 @@ pub fn update_stake_pool_balance(
     })
 }
 
-/// Creates a 'Deposit' instruction.
+/// Creates instructions required to deposit into a stake pool, given a stake
+/// account owned by the user
 pub fn deposit(
```

Comment:
wydt about  `deposit()` and `deposit_private()` functions, and removing `deposit_authority_must_sign: bool` from `deposit()`?  `deposit_private()` implies the deposit authority will sign, `deposit()` doesn't.  Just a thought, I don't feel strongly

Reply:
Or how about this:
1. `deposit()` doesn't have `stake_pool_deposit_authority`, it runs `find_deposit_authority_program_address()` and adds it automatically
2. `deposit_with_authority()` has `stake_pool_deposit_authority`, and assumes `deposit_authority_must_sign = true` 

The cli can then select 1 or 2 depending on whether a `--deposit-authority` was provided to it by the user

Reply:
Oh I really like that idea!